File: 2015/gara3/festa/sol/sbagliata.pas
uses math;

const
    MAXN = 10240;
    MAXM = 102400;
var
    N, M, i                  : longint;
    conoscenzeA, conoscenzeB : array[0..MAXM-1] of longint;
    nNext                    : array[0..MAXN-1] of longint;
    fr, fw                   : text;

function invita(N, M: longint; var da, a: array of longint): longint;
var t, i : longint;
begin
    for i:=0 to M-1 do begin
        inc(nNext[da[i]]);
        inc(nNext[a[i]]);
    end;
    invita := N;
    for i:=0 to N-1 do
        if (nNext[i] < 2) then begin
            dec(invita);
        end;
end;


begin
{$ifdef EVAL}
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
{$else}
    fr := input;
    fw := output;
{$endif}
    reset(fr);
    rewrite(fw);
    readln(fr, N, M);
    for i:=0 to M-1 do
        readln(fr, conoscenzeA[i], conoscenzeB[i]);

    writeln(fw, invita(N, M, conoscenzeA, conoscenzeB));
    close(fr);
    close(fw);
end.

File: 2015/gara4/calcolatrice/sol/sbagliata.cpp
#include <stdio.h>
#include <assert.h>

int rimedia(int N) {
    int T = N, M = 0, c;

    while (T > 0) {
        c = T % 10;
        M = M * 10 + c;
        T /= 10;
    }
    return M < N ? M : N;
}


int main() {
    FILE *fr, *fw;
    int N;
    
#ifdef EVAL
    fr = fopen("input.txt", "r");
    fw = fopen("output.txt", "w");
#else
    fr = stdin;
    fw = stdout;
#endif
    assert(1 == fscanf(fr, "%d", &N));
    
    fprintf(fw, "%d\n", rimedia(N));
    fclose(fr);
    fclose(fw);
    return 0;
}
